vllm/v1/worker/gpu_model_runner.py

```diff
@@ -2786,6 +2796,32 @@ def load_model(self, eep_scale_up: bool = False) -> None:
                 self.model = UBatchWrapper(self.model, self.vllm_config,
                                            CUDAGraphMode.NONE, self.device)
 
+    def _get_eagle3_aux_layers_from_config(self) -> Optional[tuple[int, ...]]:
+        """Extract Eagle3 auxiliary layer IDs from speculative config.
+
+        Returns:
+            Tuple of layer indices if found in draft model config,
+            None otherwise.
+        """
+        if not (self.speculative_config
+                and self.speculative_config.draft_model_config):
+            return None
+
+        try:
+            hf_config = self.speculative_config.draft_model_config.hf_config
+            if not hasattr(hf_config, 'eagle_aux_hidden_state_layer_ids'):
```

Comment:
Is this the standard for eagle3 checkpoints?
Is this what we're expecting for speculators?

Reply:
yes to both!